pkg/provider/provider.go

```diff
@@ -973,3 +1041,20 @@ func canonicalNamespace(ns string) string {
 
 // deleteResponse causes the resource to be deleted from the state.
 var deleteResponse = &pulumirpc.ReadResponse{Id: "", Properties: nil}
+
+// parseKubeconfigString takes a raw kubeconfig (YAML or JSON) string and attempts to unmarshal it into
+// a Config struct.
+func parseKubeconfigString(kubeconfig string) (*clientapi.Config, error) {
+	// If the kubeconfig is loaded as a config value by Pulumi,
+	// the string can be `"<nil>"` rather than simply `""`.
```

Comment:
Interesting. Do you know in which cases this happens?

Reply:
It was happening when I called `.String()` on a field that didn't exist in the RPC object. For example,
```go
// Shows up as `"<nil>"` if `kubeconfig` wasn't set on the provider args.
kubeconfig := news["kubeconfig"].String()
```

Reply:
I figured out how to handle this more cleanly with the `IsNull` and `StringValue` methods.